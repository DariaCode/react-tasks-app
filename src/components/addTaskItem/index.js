import AddTaskItem from "./addTaskItem";
export { AddTaskItem };