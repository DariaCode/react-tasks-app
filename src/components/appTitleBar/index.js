import AppTitleBar from "./appTitleBar";
export { AppTitleBar };