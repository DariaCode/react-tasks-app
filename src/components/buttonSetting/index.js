import { ButtonSetting } from './buttonSetting';
export { ButtonSetting }