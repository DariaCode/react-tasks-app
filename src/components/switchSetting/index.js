import { SwitchSetting } from './switchSetting';
export { SwitchSetting }