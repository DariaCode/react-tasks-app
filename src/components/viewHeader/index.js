import { ViewHeader } from "./viewHeader";
export  { ViewHeader };