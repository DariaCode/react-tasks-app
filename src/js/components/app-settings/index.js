import AppSettings from './app-settings';
export { AppSettings }