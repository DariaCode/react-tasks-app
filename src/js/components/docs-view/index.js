import { DocsView } from "./docs-view";
export { DocsView };