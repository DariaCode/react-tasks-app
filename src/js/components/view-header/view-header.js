import React from "react";
import { Title } from "../title";
import { TextLink } from "../text-link";

export const ViewHeader = props => {
	return(
		<div className={`view-header ${props.classes}`}>
			<TextLink link={props.link}>{props.linkLabel}</TextLink>
			<Title element={props.element} classes="size-lg">{props.title}</Title>
		</div>
	);
}

ViewHeader.defaultProps = {
	classes: "",
	linkLabel: "",
	title: "",
	link: false
}