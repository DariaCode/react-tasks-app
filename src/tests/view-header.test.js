import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json'
import { ViewHeader } from '../js/components/view-header';

describe('ViewHeader', () => {

	it('should render without crashing', () => {
		shallow(
			<ViewHeader
				linkLabel="Test link label"
				link="/"
				title="Test title"
			/>
		);
	})

});

describe('ViewHeader - default render', () => {

	let component;

	beforeAll( () => {
		component = shallow(
			<ViewHeader />
		);
	});

	it('should match snapshot', () => {
		expect(toJson(component)).toMatchSnapshot();
	});

	it('should render the root node as a "div" tag', () => {
		expect(component.type()).toBe('div');
	});

	it('should have the correct class name applied to the root node', () => {
		expect(component.at(0).hasClass('view-header')).toEqual(true);
	});

	it('should contain one child React component equal to "<TextLink />"', () => {
		expect(component.children('TextLink').length).toEqual(1);
	});

	it('should contain one child React component equal to "<Title />"', () => {
		expect(component.children('Title').length).toEqual(1);
	});

	it('should set a class of "size-lg" to the "<Title />" child component', () => {
		expect(component.children('Title').prop('classes')).toBe('size-lg');
	});

	afterAll( () => {
		component.unmount();
	});

});

describe('ViewHeader - additional classes', () => {

	let component;

	beforeAll( () => {
		component = shallow(
			<ViewHeader
				classes="test-view-header-class"
			/>
		);
	});

	it('should output the correct class name when provied via "classes" prop', () => {
		expect(component.at(0).hasClass('test-view-header-class')).toEqual(true);
	});

	afterAll( () => {
		component.unmount();
	});

});

describe('ViewHeader - integration <TextLink />', () => {

	let component;

	beforeAll( () => {
		component = shallow(
			<ViewHeader
				linkLabel="Test link label"
				link="/"
			/>
		);
	});

	it('should pass the "linkLabel" prop value to the "<TextLink />" child component child prop', () => {
		expect(component.children('TextLink').prop('children')).toBe('Test link label');
	});

	it('should pass the "link" prop value to the "<TextLink />" child component link prop', () => {
		expect(component.children('TextLink').prop('link')).toBe('/');
	});

	afterAll( () => {
		component.unmount();
	});

});

describe('ViewHeader - integration <TextLink />', () => {

	let component;

	beforeAll( () => {
		component = shallow(
			<ViewHeader
				title="Test title"
				element="h1"
			/>
		);
	});

	it('should pass the "title" prop value to the "<Title />" child component child prop', () => {
		expect(component.children('Title').prop('children')).toBe('Test title');
	});

	it('should pass the "element" prop value to the "<Title />" child component elemement prop', () => {
		expect(component.children('Title').prop('element')).toBe('h1');
	});

});