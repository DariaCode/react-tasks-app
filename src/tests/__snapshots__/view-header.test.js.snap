// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewHeader - default render should match snapshot 1`] = `
<div
  className="view-header "
>
  <TextLink
    classes=""
    external={false}
    link={false}
  />
  <Title
    classes="size-lg"
    element="p"
  />
</div>
`;
